src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -209,6 +222,18 @@ public IDataLoader<IMultiStreamSource> Load(Stream stream)
             }
         }
 
+        /// <summary>
+        /// Load the model from the file.
+        /// </summary>
+        /// <param name="modelPath">Path to model.</param>
+        /// <returns>A model of type <see cref="CompositeDataLoader{IMultiStreamSource, ITransformer}"/> containing the loader
+        /// and the transformer chain.</returns>
+        public IDataLoader<IMultiStreamSource> Load(string modelPath)
```

Comment:
This API seems wrong to me.

‘mlContext.Model.Load(“path”)’ returning an ‘IDataLoader’ doesn’t seem right.

Can/should it be renamed to ‘LoadDataLoader’ or similar?

Reply:
Yes, probably -- overloads that return fundamentally different objects strike me as confusing.

This would affect not just this added method but the one above, just to be clear.

---
In reply to: [266724059](https://github.com/dotnet/machinelearning/pull/3008#discussion_r266724059) [](ancestors = 266724059)

Reply:
Done.